frontend/src/js/controllers/profileCtrl.js

```diff
@@ -169,6 +175,193 @@
+                                $rootScope.notify("error", error);
+                            }
+                        }
+
+                    }
+                };
+
+                utilities.sendRequest(parameters);
+
+            } else {
+                $mdDialog.hide();
+                $state.reload();
+
+            }
+        };
+
+        // toggle old password visibility
+        vm.toggleOldPasswordVisibility = function() {
+            $rootScope.canShowOldPassword = !$rootScope.canShowOldPassword;
+        };
+
+        // toggle new password visibility
+        vm.toggleNewPasswordVisibility = function() {
+            $rootScope.canShowNewPassword = !$rootScope.canShowNewPassword;
+        };
+
+        // toggle new password again visibility
+        vm.toggleNewConfirmVisibility = function() {
+            $rootScope.canShowNewConfirmPassword = !$rootScope.canShowNewConfirmPassword;
+        };
+
+        // function to change password
+        vm.changePassword = function(resetconfirmFormValid) {
+            if(resetconfirmFormValid){
+            var parameters = {};
+            parameters.url = 'auth/password/change/';
+            parameters.method = 'POST';
+            parameters.data = {
+                "old_password": vm.user.old_password,
+                "new_password1": vm.user.new_password1,
+                "new_password2": vm.user.new_password2,
+                "uid": $state.params.user_id,
+            };
+            parameters.token = userKey;
+                parameters.callback = {
+                    onSuccess: function() {
+                        vm.user.error = false;
+                        $rootScope.notify("success", "Your password has been changed successfully!");
+                        $state.go('web.profile.AuthToken');
+                    },
+                    onError: function(response) {
+                        vm.user.error = "Failed";
+                        vm.isFormError = true;
+                        var oldpassword_valid ,password1_valid, password2_valid;
+                        try {
+                            oldpassword_valid = typeof(response.data.old_password) !== 'undefined' ? true : false;
+                            password1_valid = typeof(response.data.new_password1) !== 'undefined' ? true : false;
+                            password2_valid = typeof(response.data.new_password2) !== 'undefined' ? true : false;
+                            if (oldpassword_valid) {
+                                vm.FormError = Object.values(response.data.old_password).join(" ");
+                            }else if (password1_valid) {
+                                vm.FormError = Object.values(response.data.new_password1).join(" ");
+                            } else if (password2_valid) {
+                                vm.FormError = Object.values(response.data.new_password2).join(" ");
+                            }
+                        } catch (error) { 
```

Comment:
Why was this method added here?
How was it handled previously?

Reply:
These all new functions were previously in a seprate updateprofile and changepassword files, since we are now working on a single page with most of the stuff. i added them here.